ui/src/app/machines/views/MachineDetails/MachineNetwork/AddInterface/AddInterfaceFormFields/AddInterfaceFormFields.tsx

```diff
@@ -0,0 +1,134 @@
+import { Col, Input, Row } from "@canonical/react-components";
+import { useFormikContext } from "formik";
+import { useSelector } from "react-redux";
+
+import type { AddInterfaceValues } from "../types";
+
+import FabricSelect from "app/base/components/FabricSelect";
+import FormikField from "app/base/components/FormikField";
+import LinkModeSelect from "app/base/components/LinkModeSelect";
+import MacAddressField from "app/base/components/MacAddressField";
+import SubnetSelect from "app/base/components/SubnetSelect";
+import TagField from "app/base/components/TagField";
+import VLANSelect from "app/base/components/VLANSelect";
+import fabricSelectors from "app/store/fabric/selectors";
+import { NetworkLinkMode } from "app/store/machine/types";
+import subnetSelectors from "app/store/subnet/selectors";
+import type { Subnet } from "app/store/subnet/types";
+
+const fieldOrder = ["fabric", "vlan", "subnet", "mode", "ip_address"];
+
+const AddInterfaceFormFields = (): JSX.Element | null => {
+  const fabrics = useSelector(fabricSelectors.all);
+  const subnets = useSelector(subnetSelectors.all);
+  const { setFieldValue, values } = useFormikContext<AddInterfaceValues>();
+  const setField = <N extends keyof AddInterfaceValues>(
```

Comment:
Do you think this should just be `resetFollowingFields` and only handle the resetting portion of this function? I don't think it's the worst thing to explicitly call two separate functions that accomplish two separate things, plus `setField` is quite vague.

Reply:
Sure.